subsys/net/lib/rest_client/src/rest_client.c

```diff
@@ -273,11 +273,14 @@ static void rest_client_close_connection(struct rest_client_req_context *const r
 		ret = close(req_ctx->connect_socket);
 		if (ret) {
 			LOG_WRN("Failed to close socket, error: %d", errno);
+		} else {
+			LOG_DBG("Socket with id: %d was closed", req_ctx->connect_socket);
 		}
+		resp_ctx->used_socket_is_alive = false;
```

Comment:
Most of the `resp_ctx` values are already initialized in `rest_client_do_api_call()`, only `http_status_code` and `used_socket_is_alive` are left uninitialized. I would move this line there and also set `http_status_code` to zero.